proposals/dictionary-expressions.md

```diff
@@ -402,36 +316,152 @@ Synthesized types are free to employ any strategy they want to implement the req
+Dictionary<int, string> map1 = [pair1, pair2]; // ?
+```
+
+Resolution: While cute, these capabilities are not needed for core scenarios to work.  They also raise concerns about where to draw the line wrt to what is the dictionary space and what is not.  As such, we will only allow `KeyValuePair<,>` for now.  And we will not do anything with tuples and/or other deconstructible types.  This is also something that could be relaxed in the future if there is sufficient feedback and motivation to warrant it.  This design space is withdrawn from dictionary expressions.
+
+### Question: Semantics when a type implements the *dictionary type* shape in multiple ways.
+
+What are the rules when types have multiple indexers and multiple impls of `IEnumerable<KVP<,>>`?  
+
+This concern already exists with *collection types*.  For those types, the rule is that we must have an *element type* as per the existing language rules.  This follows for *dictionary types*, along wih the rule that there must be a corresponding indexer for this *element type*.  If those hold, the type can be used a a *dictionary type*.  If these don't hold, it cannot be.
+
+## Open Questions
+
+### Question: Types that support both collection and dictionary initialization
 
-It would likely make sense to align with whatever comes out if you `foreach`ed the collection.
+C# 12 supports collection types where the element type is some `KeyValuePair<,>`, where the type has an applicable `Add()` method that takes a single argument. Which approach should we use for initialization if the type also includes an indexer?
+
+For example, consider a type like so:
+
+```c#
+public class Hybrid<TKey, TValue> : IEnumerable<KeyValuePair<TKey, TValue>>
+{
+    public void Add(KeyValuePair<TKey, TValue> pair);
+    public TValue this[TKey key] { ... }
+}
+
+// This would compile in C# 12:
+// Translating to calls to .Add.
+Hybrid<string, int> nameToAge = [someKvp];
+```
+
+Options include:
+
+1. Use applicable instance indexer if available; otherwise use C#12 initialization.
+2. Use applicable instance indexer if available; otherwise report an error during construction (or conversion?).
+3. Use C#12 initialization always.
+
+Resolution TBD.  Working group recommendation: Use applicable instance indexer only.  This ensures that everything dictionary-like is initialized in a consistent fashion.  This would be a break in behavior when recompiling.  The view is that these types would be rare.  And if they exist, it would be nonsensical for them to behave differently using the indexer versus the .Add (outside of potentially throwing behavior).
+
+### Question: Parsing ambiguity
+
+Parsing ambiguity around: `[a ? [b] : c]`
+
+Working group recommendation: Use normal parsing here.  So this would be the same as `[a ? ([b]) : (c)]` (a collection expression containing a conditional expression).  If the user wants a `key_value_pair_element` here, they can write: `[(a?[b]) : c]`
```

Comment:
very low level concern.  Just want to get sign off that the direction the existing impl takes is the desired one.